### DISCUSSION TITLE
[Tips] 尽量使用精简版的 v2ray 的数据文件以加快启动速度

### DISCUSSION THREAD

**Discussion by IrineSistiana** [IrineSistiana/mosdns#258] [Ideas]
受打包机制的限制，读取 v2ray 的 dat 文件时需要将其**全部解包**，再提取用户实际需要的部分。

比如目前 v2ray 官方的 geoip.dat 大小已经 11M 了，其内部包含几乎世界全部国家的近百万 IP 段。路由等性能不是很强的平台，解包包含如此多数据的 geoip.dat 可能需要 5~10 秒甚至更多的时间。(给个参考: 标压 i7 需要 0.2 秒)。

geosite.dat 文件同理。

**解决方法:** 

- 如果使用的数据仓库提供精简包就使用精简包。
- 没有精简包，也可以自己解包: mosdns 有个小工具命令可以解包 dat 文件成独立的文本文件。

Update: 

- 大的数据文件**仅**影响加载速度，不影响内存占用和匹配速度。
- 解包的命令详见 wiki。

**Comment by rampageX** [comment]
通常匹配国内规则应该用它那个精简的 cn 版足够了：

> geoip-only-cn-private.dat（精简版 GeoIP，只包含 geoip:cn 和 geoip:private）

179k

https://github.com/Loyalsoldier/geoip

**Comment by IrineSistiana** [comment]
> 通常匹配国内规则应该用它那个精简的 cn 版足够了：
> 
> > geoip-only-cn-private.dat（精简版 GeoIP，只包含 geoip:cn 和 geoip:private）
> 
> 179k

好东西啊。之前没发现.....

**Comment by thinker007** [comment]
真是好东西，一下子内存占用正常了

**Comment by yeezylife** [comment]
> cn.dat（精简版 GeoIP，只包含 geoip:cn）

> geoip-only-cn-private.dat（精简版 GeoIP，只包含 geoip:cn 和 geoip:private）

是不是用cn.dat就行啊,那个 `geoip:private` 也用不到啊

**Comment by christaikobo** [comment]
话说其实IPIP的库是不是更好
体积更小
而且更权威些？
https://github.com/17mon/china_ip_list

**Comment by PHCSJC** [comment]
问个题外话，不单独开贴了，cache 1万条数据大概占多少内存呢？有办法看到cache的实时数吗？好做评估

**Comment by 0xffffharry** [comment]
> 话说其实IPIP的库是不是更好 体积更小 而且更权威些？ https://github.com/17mon/china_ip_list

v2ray-rules-dat 好像就是用这个库读取国内ip段的

**Comment by Slotheve** [comment]
> 受打包机制的限制，读取 v2ray 的 dat 文件时需要将其**全部解包**，再提取用户实际需要的部分。
> 
> 比如目前 v2ray 官方的 geoip.dat 大小已经 11M 了，其内部包含几乎世界全部国家的近百万 IP 段。路由等性能不是很强的平台，解包包含如此多数据的 geoip.dat 可能需要 5~10 秒甚至更多的时间。(给个参考: 标压 i7 需要 0.2 秒)。
> 
> geosite.dat 文件同理。
> 
> **解决方法:**
> 
> * 如果使用的数据仓库提供精简包就使用精简包。
> * 没有精简包，也可以自己解包: mosdns 有个小工具命令可以解包 dat 文件成独立的文本文件。

请问如何使用mosdns解包dat文件呢
